### PR TITLE
WIFIMD5, RTC, and Unk0202254

### PR DESCRIPTION
**Pull request by SethBarberee**

### CODE REVIEW

File: arm9/lib/src/RTC_internal.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+    return RtcSendPxiCommand(18);
+}
+
+u32 RTCi_WriteRawStatus2Async(void)
+{
+    return RtcSendPxiCommand(39);
+}
+
+u32 RtcSendPxiCommand(u8 command)
```
Comment (tgsm) [inline arm9/lib/src/RTC_internal.c:25]: This function is probably a `BOOL`

File: arm9/lib/src/RTC_internal.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    s32 data = command << 8 & 0x7f00;
+    if(PXI_SendWordByFifo(PXI_FIFO_TAG_RTC, data, FALSE) >= 0)
+        return 1;
+    else
+        return 0;
```
Comment (tgsm) [inline arm9/lib/src/RTC_internal.c:31]:
```suggestion
    return PXI_SendWordByFifo(PXI_FIFO_TAG_RTC, data, FALSE) >= 0;
```

File: arm9/lib/src/RTC_internal.c (hunk 1 above)
Comment (SethBarberee) [reply arm9/lib/src/RTC_internal.c]: Yup, matched as BOOL. Applied it to rest of `RTC_internal`. Thanks!

File: arm9/lib/include/RTC_internal.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#ifndef NITRO_RTC_INTERNAL_H_
+#define NITRO_RTC_INTERNAL_H_
+
+BOOL RtcSendPxiCommand(u8 command);
```
Comment (red031000) [inline arm9/lib/include/RTC_internal.h:4]: this one should be static and in the .c file

File: arm9/lib/src/RTC_internal.c [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#include "global.h"
+#include "PXI_fifo.h"
+#include "RTC_internal.h"
+
+BOOL RTCi_ReadRawDateTimeAsync(void)
```
Comment (red031000) [inline arm9/lib/src/RTC_internal.c:5]: all of these should be prefixed with ARM_FUNC

File: arm9/lib/src/RTC_internal.c [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#include "global.h"
```
Comment (red031000) [inline arm9/lib/src/RTC_internal.c:1]: don't include global.h in sdk files, these can cause some nasty issues

File: arm9/lib/src/RTC_internal.c [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+{
+    return RtcSendPxiCommand(18);
+}
+
+BOOL RTCi_WriteRawStatus2Async(void)
+{
+    return RtcSendPxiCommand(39);
+}
+
+BOOL RtcSendPxiCommand(u8 command)
```
Comment (red031000) [inline arm9/lib/src/RTC_internal.c:25]: this one should be static

File: arm9/src/WIFI_stubsmd5_dummy.c [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#include "global.h"
```
Comment (red031000) [inline arm9/src/WIFI_stubsmd5_dummy.c:1]: again don't include global.h here

File: arm9/lib/include/RTC_internal.h [hunk 8]
```diff
@@ -0,0 +1,10 @@
+#ifndef NITRO_RTC_INTERNAL_H_
+#define NITRO_RTC_INTERNAL_H_
+
```
Comment (red031000) [inline arm9/lib/include/RTC_internal.h:3]: include nitro/types.h here

File: arm9/lib/include/RTC_internal.h [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#ifndef NITRO_RTC_INTERNAL_H_
```
Comment (red031000) [inline arm9/lib/include/RTC_internal.h:1]: please be consistent with header guard naming

File: include/WIFI_stubsmd5_dummy.h [hunk 10]
```diff
@@ -0,0 +1,12 @@
+#ifndef POKEDIAMOND_WIFI_STUBSMD5_DUMMY_H
+#define POKEDIAMOND_WIFI_STUBSMD5_DUMMY_H
+
```
Comment (red031000) [inline include/WIFI_stubsmd5_dummy.h:3]: include nitro/types.h

File: arm9/src/unk_02022504.c [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+{
+    return SavArray_get(sav2, 32);
+}
+
+THUMB_FUNC u8 *LoadHallOfFame(struct SaveBlock2 *sav2, u32 heap_id, int *ret_p)
+{
+    return ReadSaveFileFromFlash(sav2, heap_id, 0, ret_p);
+}
+
+THUMB_FUNC int *SaveHallOfFame(struct SaveBlock2 *sav2, u8 *data)
```
Comment (red031000) [inline arm9/src/unk_02022504.c:29]: pretty sure this should return s32 not int

File: include/unk_02022504.h [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+void *FUN_02022504(struct SaveBlock2 *sav2);
+struct PCStorage *GetStoragePCPointer(struct SaveBlock2 *sav2);
+void *FUN_0202251C(struct SaveBlock2 *sav2);
+void *FUN_02022528(struct SaveBlock2 *sav2);
+u8 *LoadHallOfFame(struct SaveBlock2 *sav2, u32 heap_id, int *ret_p);
+int *SaveHallOfFame(struct SaveBlock2 *sav2, u8 *data);
+
+#endif //POKEDIAMOND_UNK_02022504_H
+
```
Comment (red031000) [inline include/unk_02022504.h:15]: extra blank line

File: include/WIFI_stubsmd5_dummy.h [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+#include "dgt.h"
+
+void MD5Final(u8 *digest, struct DGTHash1Context *context);
+void MD5Update(struct DGTHash1Context *context, u8 *input, u32 length);
+void MD5Init(struct DGTHash1Context *context);
+
+
```
Comment (red031000) [inline include/WIFI_stubsmd5_dummy.h:10]: extra blank line

**Review by red031000 — CHANGES_REQUESTED** [review]

**Review by red031000 — APPROVED** [review]

**Pull request merged**